logic/standalone/conflict.py

```diff
@@ -0,0 +1,209 @@
+'''
+<SUMMARY TBA>
+
+USAGE EXAMPLE:
+    raw_dict = conflict.CheckConflicts(playdo, _LIST_LIGHTING_OBJ)
+    pruned_dict = conflict.PruneConflicts(playdo, conflict_dictionary)
+    conflict.FixConflicts(playdo, pruned_dict)
+'''
+
+import logic.common.log_utils as log
+import logic.common.tiled_utils as tiled_utils
+
+#--------------------------------------------------#
+'''Variables'''
+
+
+
+
+
+#--------------------------------------------------#
+'''Public Functions'''
+
+def CheckConflicts(playdo, list_scan_obj):
+	''' Task 1 - Check if there is any conflict '''
+	log.Info("\n--- Checking for sort value conflicts ---\n")
+
+	# Create the dictionary with each unique sort value as keys
+	dict = {}
+	for scanned_obj in list_scan_obj:
+		curr_obj_list = playdo.GetAllObjectsWithName(scanned_obj)
+		log.Info( f'# of \"{scanned_obj}\": {len(curr_obj_list)}' )
+		for obj in curr_obj_list:
+			curr_string = tiled_utils.GetSort(obj)
+			if curr_string == None: continue
+			if not curr_string in dict:
+				dict[curr_string] = []
+			dict[curr_string].append(obj)
+
+
+	# Print the result
+	max_name_len = len( max(list_scan_obj, key = len) )    # Length of longest name in list
+	log.Info('')
+	log.Info('~~Note: Sort Layers with only 1 element are skipped~~')
+	for key, value in dict.items():
+		if len(value) == 1: continue
+		log.Extra( '--------------------------------------------------' )
+		log.Info( f"{key} has {len(value)} elements" )
+		for obj in value:
+			log.Extra( f'    {PrintObjInfo(obj, GetParentName(obj, playdo), max_name_len)}' )
+	log.Extra( '--------------------------------------------------' )
+
+
+	log.Must("\n--- Finished checking conflicts! ---\n")
+	return dict
+
+
+
+
+
+#--------------------------------------------------#
+'''Utility'''
+
+
+def GetParentName(obj, playdo):
+	parent = GetParent(obj, playdo)
+	if parent == None: return '...'
+	return parent.get('name')[8:]    # Remove beginning, i.e. 'objects_'
+
+
+# TODO relocate to tiled_utils?
+# TODO List comprehension?
+def GetParent(obj, playdo):
+	'''Return the name of the layer that the object is in'''
+	root = playdo.level_root
+
+	# Check all object layers that are in 0~1 layer of folder
+	all_objectgroup = root.findall('objectgroup')
+	for folder in root.findall('group'):
+		all_objectgroup += folder.findall('objectgroup')
+
+	# Check which objectgroup contains the obj
+	for group in all_objectgroup:
+		for child in group:
+			if child == obj: return group
+	return None
```

Comment:
Pretty good attempt! It definitely gets the job done.

Though it's sorta wasteful with how much we're iterating to re-get each parent. Looking online, I found this which does the trick nicely:
https://stackoverflow.com/questions/2170610/access-elementtree-node-parent-node

It's a one-liner:
`parent_map = {c: p for p in tree.iter() for c in p}`

Construct the parent_map once and cache it. After that, you can get the parent for each object with a simple dictionary fetch. Give it a shot.

If you get stuck, here's an example with some wider context on how it'd be used ( https://imgur.com/WwUISPf)